### PR TITLE
Funnel improvements - Graph type, identical steps, UI updates

### LINKED ISSUE #5088: Explain how identical funnel steps work in the UI
## Is your feature request related to a problem?

Currently users get confused by the way identical funnel steps work. When two or more steps are identical _one after another_, they sometimes expect there to be no dropoff from identical step 1 to identical step 2.

But the way it really works (shown below) is much more practical:
What PostHog does instead, is that out of the set of people who performed the step once (making up step 1), for step 2 it counts people who performed that step for a second time in the funnel window. This way it's possible to construct funnels requiring a user to perform a specific action _n_ times before carrying on with the rest of the funnel.

Funnel

## Describe the solution you'd like

If a step is identical to the previous one, it should have a tooltip (over an ℹ️ icon maybe) explaining why there's a dropoff.

## Additional context

See #5046 or https://posthog.slack.com/archives/C0113360FFV/p1626110518064100.

### LINKED ISSUE #5235: Improve funnels graph type selector
I'm pretty unconvinced with the current graph type selector component. It feels too cramped, hard to communicate what each option is easily, and doesn't play nice in some screen dimensions. Here's a proposal to improve it. Thoughts @clarkus ?

**Current**

**Proposal**

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR implements a few improvements to funnels and insights in general:

1. We now show information about duplicate and sequential steps (closes #5088)

2. Filters & breakdown are now split into a separate card following the mockups for this.

3. The "Add filter", "Add breakdown" & "Add formula" buttons across all insights are now links instead of a default button.

4. We implement a brand new graph type selector that looks good even on narrow screens (closes #5235).

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by paolodamico** [comment]
@clarkus one thing I'm not sure about is how the new link type button displays in certain contexts, see below (spacing looks strange), wdyt?

**Comment by clarkus** [comment]
These changes look good to me. Regarding the link placement, in this type of layout, I generally try to align with the input value. It's like that link has the same box properties as the form above it, just without color values. 
Screen Shot 2021-07-30 at 3 20 54 PM

**Review by clarkus — APPROVED** [review]

**Comment by paolodamico** [comment]
@alexkim205 could you help me out with a code review and maybe just address this aligning improvement? I'm swamped in other issues right now

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelSecondaryTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    propertyFilters={filters.properties || []}
+                    onChange={(anyProperties) => {
+                        setFilters({
+                            properties: anyProperties.filter(isValidPropertyFilter),
+                        })
+                    }}
+                />
+                <TestAccountFilter filters={filters} onChange={setFilters} />
+                {/* TODO */}
+                {(true || clickhouseFeaturesEnabled) && filters.funnel_viz_type === FunnelVizType.Steps && (
```
Comment (alexkim205) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelSecondaryTabs.tsx:41]: Did you mean to remove this `true`?

**Comment by alexkim205** [comment]
- [x] Added an arrow to mimic our property filter nesting in other insight graphs
- [x] Aligned 'Add new filter' to text

CC @clarkus @paolodamico 

Screen Shot 2021-08-02 at 10 59 29 AM

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelSecondaryTabs.tsx (hunk 1 above)
Comment (paolodamico) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelSecondaryTabs.tsx:41]: Pff good catch, thanks Alex!

**Pull request merged**